src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -142,6 +142,8 @@ private async Task<bool> UseGoCliToScanAsync(string location, ISingleFileCompone
         {
             this.Logger.LogError("Go CLI command \"go list -m -json all\" failed with error: {GoDependenciesProcessStdErr}", goDependenciesProcess.StdErr);
             this.Logger.LogError("Go CLI could not get dependency build list at location: {Location}. Fallback go.sum/go.mod parsing will be used.", location);
+            record.DidGoCliCommandFail = true;
+            record.GoCliCommandError = goDependenciesProcess.StdErr;
```

Comment:
There's a chance this is too long and will cause JSON truncation - will keep an eye on that.